File: js/src/sorts/insertionSort.js
function InsertionSort() {

}

InsertionSort.prototype = {
  sort: function(arr) {
    var input = arr.slice(0);

    for(var i = 0; i < input.length - 1; i++) {
      var j = i + 1;

      // swap the number at index i + 1 back until it's in the right place
      while(input[j] < input[j - 1]) {
        util.swap(input, j, j - 1);
        j--;
      }
    }

    return input;
  }
}

File: ruby/src/sorts/insertion_sort.rb
module InsertionSort
    def InsertionSort.sort(arr)
        # copy input array
        out = arr.map { |x| x }

        for i in (1...out.length)
            j = i

            # as long as our current value is less than the value before it
            while j > 0 && out[j] < out[j - 1]
                # swap the values
                out[j], out[j - 1] = out[j - 1], out[j]
                j = j - 1
            end
        end

        out
    end
end
